lxd/network/openvswitch/ovn.go

```diff
@@ -2047,3 +2082,192 @@ func (o *OVN) LogicalRouterPeeringDelete(opts OVNRouterPeering) error {
 
 	return nil
 }
+
+// GetHardwareAddress gets the mac address of the ovn managed network.
+func (o *OVN) GetHardwareAddress(networkID int64) (string, error) {
+	nameFilter := fmt.Sprintf("name=lxd-net%d-lr-lrp-ext", networkID)
```

Comment:
O could be wrong but I think the ovn client library doesn't previously have the concept of network Id as is supposed to not be tightly coupled to lxds network concepts.

Reply:
So it looks like the network id does make it into the OVN name via the following route:

`n.Create` calls `n.setup` which calls `client.ChassisGroupAdd` with `n.getChassisGroupName()` as the OVNChassisgroup argument. `n.getChassisGroupName` then calls `n.getNetworkPrefix` and finally `acl.OVNNetworkPrefix(n.id)`, which returns `lxd-net<id>`.

So I guess with the current version of ovn we can expect to find names that include the chassis group name. Of course this might change, but I'm not sure how to filter the results without it.

What I will do at lease in the meantime is pass in the chassis group name as an argument and use that to create the filter instead.

Reply:
@markylaing yes that was my point, that networkID (as a standalone concept) is not used in the `openvswitch` package directly, but rather OVS/OVN object names are used (which LXD then might choose to embed its networkID into).

My thinking with the `openvswitch` package was to make it a fairly low-level package that deals with configuring/accessing OVN/OVS, rather than implementing LXD's concepts on top. 

So yes, the approach you're using now of passing in the OVN object names to these functions is preferred. Thanks